reports-scheduler/src/test/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportDefinitionIT.kt

```diff
@@ -0,0 +1,303 @@
+
+        reportDefinitionRequest = constructReportDefinitionRequest(
+            """
+                "trigger":{
+                    "triggerType":"Download"
+                },
+            """.trimIndent()
+        )
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+        reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(3000)
+
+        val reportDefinitionsResponse = executeRequest(
+            RestRequest.Method.GET.name,
+            "$BASE_REPORTS_URI/definitions",
+            "",
+            RestStatus.OK.status
+        )
+        Assert.assertEquals(2, reportDefinitionsResponse.get("totalHits").asInt)
+        Thread.sleep(100)
+
+        val newName = "updated_report"
+        reportDefinitionRequest = constructReportDefinitionRequest(name = newName)
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.PUT.name,
+            "$BASE_REPORTS_URI/definition/$reportDefinitionId",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+        Assert.assertEquals(reportDefinitionId, reportDefinitionResponse.get("reportDefinitionId").asString)
+        Thread.sleep(100)
+
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.GET.name,
+            "$BASE_REPORTS_URI/definition/$reportDefinitionId",
+            "",
+            RestStatus.OK.status
+        )
+        Assert.assertEquals(
+            reportDefinitionId,
+            reportDefinitionResponse.get("reportDefinitionDetails").asJsonObject.get("id").asString
+        )
+        Assert.assertEquals(
+            newName,
+            reportDefinitionResponse.get("reportDefinitionDetails").asJsonObject.get("reportDefinition").asJsonObject.get(
+                "name"
+            ).asString
+        )
+        Thread.sleep(100)
+
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.DELETE.name,
+            "$BASE_REPORTS_URI/definition/$reportDefinitionId",
+            "",
+            RestStatus.OK.status
+        )
+        Assert.assertEquals(reportDefinitionId, reportDefinitionResponse.get("reportDefinitionId").asString)
+        Thread.sleep(100)
+
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.GET.name,
+            "$BASE_REPORTS_URI/definition/$reportDefinitionId",
+            "",
+            RestStatus.NOT_FOUND.status
+        )
+        validateErrorResponse(reportDefinitionResponse, RestStatus.NOT_FOUND.status)
+        Thread.sleep(100)
+    }
+
+    fun `test invalid get, update, delete report definition`() {
+        var reportDefinitionResponse = executeRequest(
```

Comment:
since no report definitions created this test may not cover some edge condition. may be we can create few report definitions before testing invalid ids?

Reply:
merged into test function above so there are existing report definitions